app/controllers/products_controller.rb

```diff
@@ -0,0 +1,51 @@
+class ProductsController < ApplicationController
+
+	def index
+		@products = Product.order(id: :asc)
+		render :index
+	end
+
+	def show
+    @vendor = Vendor.find(params[:id])
+    render :detail_view
+  end
```

Comment:
The problem with indents could have to do with your text editor using tabs not spaces. It's better to use spaces because the tab default on GitHub is 8 spaces which can lead to some big indents. That may or may not be the problem :)